File: tutorials/testing/python/shape_collection_textalign.py
from bridges import *
from bridges.symbol_group import *
from bridges.text import *


def anchor_test(alignLR, alignTB):

    group = SymbolGroup()

    xaxis = Polyline()
    xaxis.add_point(-50.0, 0.0)
    xaxis.add_point(50.0, 0.0)
    xaxis.stroke_color = "black"
    xaxis.stroke_width = 1.0
    group.add_symbol(xaxis)

    yaxis = Polyline()
    yaxis.add_point(0.0, -50.0)
    yaxis.add_point(0.0, 50.0)
    yaxis.stroke_color = "black"
    yaxis.stroke_width = 1.0
    group.add_symbol(yaxis)

    t = Text()
    t.font_size = 10.0
    t.stroke_width = 1.0
    t.text = "Something"
    t.stroke_color = "black"
    t.set_anchor_alignment(alignLR, alignTB)
    group.add_symbol(t)

    t2 = Text()
    t2.font_size = 10.0
    t2.stroke_width = 1.0
    t2.text = alignLR + " " + alignTB
    t2.stroke_color = "black"
    t2.set_anchor_location(0.0, -50.0)
    t2.set_anchor_alignment("middle", "top")
    group.add_symbol(t2)

    return group

def main():

    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    sc = SymbolCollection()

    sc.setviewport(-100, 400, -100, 700)

    group = anchor_test("left", "bottom")
    group.translate(0,0)
    sc.add_symbol(group)

    group = anchor_test("middle", "bottom")
    group.translate(120, 0)
    sc.add_symbol(group)

    group = anchor_test("right", "bottom")
    group.translate(240, 0)
    sc.add_symbol(group)

    group = anchor_test("left", "embottom")
    group.translate(0, 150)
    sc.add_symbol(group)

    group = anchor_test("middle", "embottom")
    group.translate(120, 150)
    sc.add_symbol(group)

    group = anchor_test("right", "embottom")
    group.translate(240, 150)
    sc.add_symbol(group)

    group = anchor_test("left", "emtop")
    group.translate(0, 450)
    sc.add_symbol(group)

    group = anchor_test("middle", "emtop")
    group.translate(120, 450)
    sc.add_symbol(group)

    group = anchor_test("right", "emtop")
    group.translate(240, 450)
    sc.add_symbol(group)

    group = anchor_test("left", "top")
    group.translate(0, 600)
    sc.add_symbol(group)

    group = anchor_test("middle", "top")
    group.translate(120, 600)
    sc.add_symbol(group)

    group = anchor_test("right", "top")
    group.translate(240, 600)
    sc.add_symbol(group)

    bridges.set_data_structure(sc)

    bridges.visualize()

if __name__ == "__main__":
    main()

File: tutorials/testing/java/shape_collection_textalign.java
import bridges.connect.Bridges;
import bridges.base.SymbolCollection;
import bridges.base.Rectangle;
import bridges.base.Circle;
import bridges.base.Text;
import bridges.base.Polyline;
import bridges.base.Polygon;
import bridges.base.Color;
import bridges.base.SymbolGroup;


public class shape_collection_textalign {

    public static SymbolGroup anchorTest(String alignLR, String alignTB) {
	SymbolGroup group = new SymbolGroup();

	// draw axes
	Polyline xaxis = new Polyline();
	xaxis.addPoint(-50.0f, 0.0f);
	xaxis.addPoint(50.0f, 0.0f);
	xaxis.setStrokeColor("black");
	xaxis.setStrokeWidth(1.0f);
	group.addSymbol(xaxis);

	Polyline yaxis = new Polyline();
	yaxis.addPoint(0.0f, -50.0f);
	yaxis.addPoint(0.0f, 50.0f);
	yaxis.setStrokeColor("black");
	yaxis.setStrokeWidth(1.0f);
	group.addSymbol(yaxis);


	//sample text
	Text l = new Text();
	l.setFontSize(10f);
	l.setStrokeWidth(1.0f);
	l.setText("Something");
	l.setStrokeColor("black");
	l.setAnchorAlignment(alignLR, alignTB);
	group.addSymbol(l);

	//title
	Text t = new Text();
	t.setFontSize(10f);
	t.setStrokeWidth(1.0f);
	t.setText(alignLR +" "+ alignTB);
	t.setStrokeColor("black");
	t.setAnchorLocation(0.0f, -50.0f);
	t.setAnchorAlignment("middle", "top");
	group.addSymbol(t);

	return group;

    }
    
	public static void main(String[] args) throws Exception {

		//create Bridges object
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");
		// title, description
		bridges.setTitle("Symbol Collection");
		bridges.setDescription("Red square, green circle, magenta horizontal and vertical lines, "
			+	"and a test label with a purple outline.");

		// create some symbols and add to symbol collection
		SymbolCollection sc = new SymbolCollection();

		sc.setViewport(-100, 400,
			       -100, 700
			       );
		
		{ 
		    SymbolGroup group = anchorTest("left", "bottom");
		    group.translate(0,0);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("middle", "bottom");
		    group.translate(120,0);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("right", "bottom");
		    group.translate(240,0);
		    sc.addSymbol(group);
		}
		

		{ 
		    SymbolGroup group = anchorTest("left", "embottom");
		    group.translate(0, 150);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("middle", "embottom");
		    group.translate(120, 150);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("right", "embottom");
		    group.translate(240, 150);
		    sc.addSymbol(group);
		}


		{ 
		    SymbolGroup group = anchorTest("left", "middle");
		    group.translate(0, 300);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("middle", "middle");
		    group.translate(120, 300);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("right", "middle");
		    group.translate(240, 300);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("left", "emtop");
		    group.translate(0, 450);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("middle", "emtop");
		    group.translate(120, 450);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("right", "emtop");
		    group.translate(240, 450);
		    sc.addSymbol(group);
		}


		{ 
		    SymbolGroup group = anchorTest("left", "top");
		    group.translate(0, 600);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("middle", "top");
		    group.translate(120, 600);
		    sc.addSymbol(group);
		}

		{ 
		    SymbolGroup group = anchorTest("right", "top");
		    group.translate(240, 600);
		    sc.addSymbol(group);
		}
		
		// set visualizer type
		bridges.setDataStructure(sc);

		// visualize the JSON and Collection
		bridges.visualize();
	}
}

